File: DatingApp/DatingApp/Entities/Report.cs
namespace DatingApp.Entities
{
    public class Report
    {
        public int Id { get; set; }

        public required string ReporterId { get; set; }
        public Member Reporter { get; set; } = null!;

        public required string ReportedUserId { get; set; }
        public Member ReportedUser { get; set; } = null!;

        public required ReportReason Reason { get; set; }
        public string? Description { get; set; }
        public string Status { get; set; } = "Pending";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: react-client/src/types/report.ts
export type Report = {
    id: number;
    reporterId: string;
    reporter: { displayName: string };
    reportedUserId: string;
    reportedUser: { displayName: string };
    reason: number;
    description?: string;
    status: string;
    createdAt: string;
};

export const ReasonLabels: { [key: number]: string } = {
    0: 'Spam',
    1: 'Harassment',
    2: 'Inappropriate Content',
    3: 'Fake Profile',
    4: 'Other'
};